crates/bevy_picking/src/pointer.rs

```diff
@@ -309,3 +211,85 @@ impl Location {
             .unwrap_or(false)
     }
 }
+
+/// Types of actions that can be taken by pointers.
+#[derive(Debug, Clone, Copy, Reflect)]
+pub enum PointerAction {
+    /// The pointer has entered a window.
+    EnteredWindow,
+    /// The pointer has left a window.
+    LeftWindow,
+    /// A button has been pressed on the pointer.
+    Pressed {
+        /// The press direction, either down or up.
+        direction: PressDirection,
+        /// The button that was pressed.
+        button: PointerButton,
+    },
+    /// The pointer has moved.
+    Moved {
+        /// How much the pointer moved from the previous position.
+        delta: Vec2,
+    },
+    /// The pointer has been canceled. The OS can cause this to happen to touch events.
+    Canceled,
+}
+
+/// An input event effecting a pointer.
+#[derive(Event, Debug, Clone, Reflect)]
+pub struct PointerInput {
+    /// The id of the pointer.
+    pub pointer_id: PointerId,
```

Comment:
I think if we're going to stick close to `winit` and thus the w3c spec they're following, we should disambiguate between device type and device id (which we current don't expose). The spec is pretty clear that "pointer id" refers to the device id. Having multiple of the same kind of input is unusual for a gaming pc, but it can't hurt to expose that to the user, including an "Unknown" device kind so the user can simulate pointer movement themselves (or use some kind of esoteric userspace device they implement).

Reply:
I'm open to this change as future work. I view this PR as mostly fixing picking ordering, and also setting up a nice transition to the winit pointer API. We can cover this sort of thing when pointers actually land upstream.

Reply:
makes sense! just wanted to call that out as specifically our use of pointer id confused me when reviewing our code in relation to the spec